spec.html

```diff
@@ -30059,13 +30082,11 @@ <h1>Function.prototype.bind ( _thisArg_, ..._args_ )</h1>
           1. If _targetHasLength_ is *true*, then
             1. Let _targetLen_ be ? Get(_Target_, *"length"*).
             1. If _targetLen_ is a Number, then
-              1. If _targetLen_ is *+&infin;*<sub>𝔽</sub>, set _L_ to +&infin;.
-              1. Else if _targetLen_ is *-&infin;*<sub>𝔽</sub>, set _L_ to 0.
-              1. Else,
-                1. Let _targetLenAsInt_ be ! ToIntegerOrInfinity(_targetLen_).
-                1. Assert: _targetLenAsInt_ is finite.
+              1. Set _L_ to ! ToIntegerOrInfinity(_targetLen_).
+              1. If _L_ is -&infin;, set _L_ to 0.
+              1. Else if _L_ is not +&infin;, then
                 1. Let _argCount_ be the number of elements in _args_.
-                1. Set _L_ to max(_targetLenAsInt_ - _argCount_, 0).
+                1. Set _L_ to max(_L_ - _argCount_, 0).
```

Comment:
I find the original easier to follow than this refactoring. @gibson042 can you say more about why you are in favor of it?

Reply:
Basically covered at https://github.com/tc39/ecma262/pull/2917#discussion_r989509978 , but two reviewers raising the same issue has persuaded me to undo it.